core/os/android/adb/commands.go

```diff
@@ -90,6 +91,15 @@ retry:
 	return log.Err(ctx, ErrRootFailed, buf.String())
 }
 
+// UserdebugBuild returns true if the device runs an Android userdebug build
+func (b *binding) UserdebugBuild(ctx context.Context) (bool, error) {
+	output, err := b.Command("shell", "getprop", "ro.build.type").Call(ctx)
+	if err != nil {
+		return false, err
+	}
+	return (output == "userdebug"), nil
```

Comment:
Is it always going to be "userdebug"? Isn't "userdebug" a substring of the actual build type?

Reply:
The complete build string is in property `ro.build.fingerprint`, e.g. 
"google/walleye/walleye:ABC123:userdebug/dev-keys"

The `ro.build.type` is only "userdebug" or "user" (or "eng", etc)

So we should be good to go by checking `ro.build.type`